scipy/sparse/_construct.py

```diff
@@ -322,6 +339,16 @@ def identity(n, dtype='d', format=None):
     format : str, optional
         Sparse format of the result, e.g., format="csr", etc.
 
+    Returns
+    -------
+    new_matrix : sparse matrix
+        A square sparse matrix with ones on (and zeros off) the main diagonal.
+
+    See Also
+    --------
+    eye_array : Sparse array with ones on (and zeros off) the specified diagonal
+    eye : Sparse matrix with ones on (and zeros off) the specified diagonal
```

Comment:
I think these would be more useful if we mentioned how they differ from the main function that we're documenting. For example, the see also section for identity() could mention that the eye() function produces ones on a specified diagonal, not just the main diagonal.